eth/protocols/snap/sync.go

```diff
@@ -2111,15 +2169,7 @@ func (s *Syncer) processStorageResponse(res *storageResponse) {
 	// Large contracts could have generated new trie nodes, flush them to disk
 	if res.subTask != nil {
 		if res.subTask.done {
-			root := res.subTask.genTrie.Commit()
-			if root == res.subTask.root {
```

Comment:
I agree about the conclusion, but not the description

> Note, this clause is to handle a special case that a storage is divided into several chunks to retrieve, but eventually all the states are still fall within the first range(uneven storage).


Say we get 100 accounts, and `100th`, `A`  has `100` slots. Since the first packet is already filled with `99` other accounts, it only contains `50` of `A`:s slots. In this case we won't start the whole "16 parallel task resolver" (no chunks) just to fetch it, but we will issue one more request. Between them, these two requests fully deliver the trie, and in that case, we would land in this clause, originally. 

At least, that's how I understand it -- a case which happens pretty often. In this case, when we receive the first reply, we ought to not set `SkipLeftBoundary` to true. 

Anyway, we can leave it as is, but should fix it later on. The clause should be something like 
```
if root == res.subTask.root && !res.subTask.genTrie.Options().BoundariesSkipped() {
```

Reply:
You modified it now, to be `rawdb.HasStorageTrieNode(s.db, accountHash, nil, root)`. So you are checking if the root storage has been committed. Yes, that should be a pretty robust solution (although: it incurs an extra database read, and we might not have committed the data to disk yet - couldn't it still be sitting in a batch?)